### DISCUSSION TITLE
Unexpected EnumerateStereoisomers() result unless sanitize during MolFromSmiles()?

### DISCUSSION THREAD

**Discussion by someonetookmyname** [rdkit/rdkit#7197] [Q&A] [answered]
Hi,

The below code generates two isomers ('C/C=C/C' and 'C/C=C\C') from the input smiles of 'C/C=C/C', for which I'm expecting only a single isomer.

This looks to be caused by "sanitize = False" in the SmilesParserParams (if I set it to True, then I get the expected EnumerateStereoisomers result).

Here and typically, I am performing post-MolFromSmiles sanitizing (like here, https://rdkit.readthedocs.io/en/latest/Cookbook.html#using-a-different-aromaticity-model) in order to prevent certain sanitizing behaviors, but it does not recover the expected result (I still get two enumerated isomers).

Perhaps meaningful, m.Debug directly after MolFromSmiles when sanitize==False shows that there are numerous valence complaints?:
	Pre-condition Violation
	getExplicitValence() called without call to calcExplicitValence()
	
Is this behavior expected?

Do I have to sanitize during MolFromSmiles()?  If so, is there a way to adjust the granularity of the sanitizing performed during MolFromSmiles()?

Thank you for any guidance.

Best, Andrew

```
sp = Chem.SmilesParserParams()
sp.allowCXSMILES = True
sp.debugParse = False
sp.parseName = False
sp.removeHs = False
sp.sanitize = False
sp.strictCXSMILES = True

sparams = Chem.SmilesWriteParams()
sparams.doRandom = False
sparams.rootedAtAtom = -1
sparams.doIsomericSmiles = True
sparams.canonical = False
sparams.doKekule = False
sparams.allHsexplicit = False
sparams.allBondsExplicit = False

smiles = 'C/C=C/C'

m = Chem.MolFromSmiles(smiles, sp)
m.Debug()
AllChem.Compute2DCoords(m)
Chem.SanitizeMol(m,sanitizeOps=Chem.SanitizeFlags.SANITIZE_ALL^Chem.SanitizeFlags.SANITIZE_KEKULIZE^Chem.SanitizeFlags.SANITIZE_SETAROMATICITY^Chem.SanitizeFlags.SANITIZE_CLEANUP^Chem.SanitizeFlags.SANITIZE_CLEANUPCHIRALITY)
Chem.Kekulize(m, clearAromaticFlags=True)
Chem.SetAromaticity(m, Chem.AromaticityModel.AROMATICITY_MDL)

se_options = StereoEnumerationOptions(tryEmbedding=False, onlyUnassigned=True, maxIsomers=0, rand=None, unique=True, onlyStereoGroups=False)
isomers = tuple(EnumerateStereoisomers(m, options=se_options))
for i in isomers:
    print(Chem.MolToCXSmiles(i, sparams, flags = (CXSmilesFields.CX_ALL^CXSmilesFields.CX_COORDS)))

```

**Comment by greglandrum** [comment, marked as answer]
[Edited 14.03.2024]
@someonetookmyname the RDKit does not do stereochemistry perception if you don't do sanitization or remove Hs, so you have to call `Chem.AssignStereochemistry()` yourself:
```
>>> smiles = 'C/C=C/C'
>>> ps = Chem.SmilesParserParams()
>>> ps.removeHs = False
>>> ps.sanitize = False
>>> m = Chem.MolFromSmiles(smiles,ps)
>>> Chem.SanitizeMol(m)
rdkit.Chem.rdmolops.SanitizeFlags.SANITIZE_NONE
>>> Chem.AssignStereochemistry(m)
>>> from rdkit.Chem import EnumerateStereoisomers
>>> isos = list(EnumerateStereoisomers.EnumerateStereoisomers(m))
>>> len(isos)
1
```

**Reply by someonetookmyname** [reply]
Thank you for the response.

It looks like FindPotentialStereo() (instead of AssignStereochemistry()) does not generate the expected result here, and the documentation suggests that there are advantages to using FindPotentialStereo() over AssignStereochemistry().

Is there any downside to calling AssignStereochemistry() (to recover this result) and then also calling FindPotentialStereo() (to get its benefits)?

**Reply by greglandrum** [reply]
`FindPotentialStereo()` doesn't actually assign stereo information to the molecule, so it does not help here.

You might want to look into using `Chem.SetUseLegacyStereoPerception(False)` in order to switch to using the newer stereo perception code. It's more accurate than the legacy code (get some cases right that the legacy code does not), but doesn't assign the pseudo-CIP codes that the old code does. If you want actual CIP labels, you can use `rdCIPLabeler`

**Reply by someonetookmyname** [reply]
Thank you for this information.
